test/runtime/Policies.go

```diff
@@ -1393,3 +1394,101 @@ var _ = Describe("RuntimeValidatedPolicyImportTests", func() {
 		Expect(res.Output().String()).Should(ContainSubstring(allowedVerdict), "Policy trace did not contain %s", allowedVerdict)
 	})
 })
+
+var _ = Describe("RuntimeValidatedPolicyDropAllTests", func() {
+	var once sync.Once
+	var logger *logrus.Entry
+	var vm *helpers.SSHMeta
+	var dropAll bool
+
+	initialize := func() {
+		logger = log.WithFields(logrus.Fields{"test": "RuntimeValidatedPolicyDropAllTests"})
+		logger.Info("Starting")
+		vm = helpers.CreateNewRuntimeHelper(helpers.Runtime, logger)
+		areEndpointsReady := vm.WaitEndpointsReady()
+		Expect(areEndpointsReady).Should(BeTrue())
+	}
+
+	BeforeEach(func() {
+		once.Do(initialize)
+		vm.PolicyDelAll()
+		res := vm.SetPolicyEnforcement(helpers.PolicyEnforcementDefault)
+		res.ExpectSuccess("Setting policyEnforcement to default")
+	})
+
+	AfterEach(func() {
+		vm.ValidateNoErrorsOnLogs(CurrentGinkgoTestDescription().Duration)
+		if CurrentGinkgoTestDescription().Failed {
+			vm.ReportFailed()
+		}
+	})
+
+	It("DROP_ALL Ingress Policy test", func() {
+		By("Starting cilium monitor in background to filter DROP requests")
+		ctx, cancel := context.WithCancel(context.Background())
+		res := vm.ExecContext(ctx, "cilium monitor --type drop")
+		defer cancel()
+
+		By("Create an endpoint with no labels to test DROP_ALL")
+		cmdStr := "docker run -dt --name ingressContainer --net=cilium-net tgraf/netperf"
+		vm.Exec(cmdStr).ExpectSuccess("Failed to create docker container")
+
+		endpointCmd := "cilium endpoint list | grep \"waiting-for-identity\" | awk '{ print $1}'"
+		endpointID := strings.TrimSpace(vm.Exec(endpointCmd).GetStdOut())
+		endpointCmd = "cilium endpoint list | grep \"waiting-for-identity\" | awk '{ print $7}'"
+		endpointIP := strings.TrimSpace(vm.Exec(endpointCmd).GetStdOut())
+
+		By("Testing endpoint lxc_config.h for DROP_ALL")
+
+		lxcCmd := fmt.Sprintf("sudo cat /var/run/cilium/state/%s/lxc_config.h | grep 'DROP_ALL'", endpointID)
+
+		dropAll = false
+		if strings.TrimSpace(vm.Exec(lxcCmd).GetStdOut()) == "#define DROP_ALL" {
+			dropAll = true
+		}
+
+		Expect(dropAll).Should(BeTrue(), "DROP_ALL failed for ingress. lxc_config.h does not contain the DROP_ALL macro")
+
+		By("Testing ingress with ping from localhost to endpoint with no labels")
+		lxcCmd = fmt.Sprintf("ping %s", endpointIP)
+		vm.Exec(lxcCmd)
+
+		By("Testing cilium monitor drop")
+		err := res.WaitUntilMatch("drop (Policy denied (L3))")
+		Expect(err).To(BeNil(), "DROP all on egress failed.")
+
+		vm.ContainerRm("ingressContainer").ExpectSuccess("Container server cannot be deleted")
```

Comment:
Do not remove containers as part of the `It` itself. Resources should be removed in the `AfterEach` construct. This is due to the following:

1. We always want to make unconditionally that resources are cleaned up after each test run. `AfterEach` is ran whether a test succeeds or fails; an `It` exits if any `Expect` statement fails, and does not run the rest of the statements in the `It`. Since for builds of master, we run all tests even if one of them fails, this means that if the code is kept as it is in its current form, there will be leftover cruft which could potentially affect the state of other tests. Since there is another `It` which creates a container, I'd recommend wrapping each `It` within a `Context`. I can give you more information about this offline, as this is easier to explain in person.
2. If the container is removed in the `It`, then if logs are gathered, then the container will already be removed and we will have no information from the Cilium CLI for the container as it won't exist any more.

Reply:
There is a reason for not adding the container creation in the BeforeEach function. The way the container is created is different i.e we need to create a docker container without any labels and also for egress we need to create a container without labels and start ping immediately on it. Hence this code cannot be common in `BeforeEach`

Reply:
@manalibhutiyani this is why I said the following

> Since there is another It which creates a container, I'd recommend wrapping each It within a Context. I can give you more information about this offline, as this is easier to explain in person

Let's discuss offline.